src/app/service/photos/photos.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PhotosService {
+
+  constructor(private http: HttpClient) { }
+
+  getPhotos(): Promise<Photo[]> {
+    return this.http.get(`https://${environment.facebookphotosUrl}`, {
+      params: { token: environment.facebookphotosToken }
+    }).toPromise()
+    .then((response: any) =>
+    //is this the correct way to make an api call?
+      FB.api(
+        "/{photo-id}",
+        function (response) {
+            if (response && !response.error) {
+                link: link,
```

Comment:
Sorry, I don't quite understand this part.
Maybe it's a dumb question, but why did you have an object directly after if statement?
Does this return or something?

Inside the if statement I mean
```javascript
{
  link: link,
  name: name
}
```
Where does the value return to?
Is this like a state change in Angular?

Reply:
For the **main logic in the .service.ts**, I think we together with Clarence should work together on it! Cuz Im not clear as well sorry..